### DISCUSSION TITLE
It’s a partner. But a dumb one

### DISCUSSION THREAD

**Discussion by ghost** [community/community#48570] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

I’ve been using it a week or two. What the we issue seems to be is it’s language based but not context aware. 
Useless blocks of code with methods that don’t exist and would be non buildable in IntelliJ. That’s in simple code completion too. Non existent methods.

It makes so many dumb suggestions but then you get a few useful lines of code that saves you typing which is great. But on balance much more gibberish to ignore. 

I have hope for these tools but not yet, I don’t understand why they think it’s worth more than the cost of my other tools, which without IntelliJ it would be not possible!

**Comment by CiberNin** [comment]
The question is if you would pay for syntax highlighting if it cost money and there was no free alternative.
Cause that's about the amount of extra utility this adds.
If you aren't out buying ergonomic keyboards and mice, then you probably won't value what Copilot has to offer.

**Reply by ghost** [reply]
It’s less than that. I pay for IntelliJ and the code completion is far better!

**Comment by NickMaczk** [comment]
I completely agree, I've been using it for 2+ weeks, very rarely it suggests me something usable, I'd say maybe a line of valid code/day and it's always about symmetry or when I have to duplicate a part of code which I can easily do myself in a short amount of time, the rest of the time I go through the same issues as you do, it uses non-existing functions and it has 0 context awareness. I was very excited because I code 12 to 16 hours/day, but it starts to feel more like a bad trainee that I have to correct every time it opens it's mouth instead of a real partner who has something to bring to the table.

**Comment by essial** [comment]
Opposite experience here. I was working on a 6502 assembler and while I was thinking about how I wanted to write my tokenizer (sitting between two {}s) it proceeded to dump _115 lines_ of code that was about 90% accurate. About 10 minutes of adjustments later and my entire tokenizer was done and worked perfectly.

Copilot is a tool, and it takes a bit to understand how to give it proper context. But once you do, oh man does it help.

**Reply by ricklabanca** [reply]
It does hit sometimes but it's so random and does not seem to learn. I just can't understand how it generates codes with methods that don't exist. It's getting to be noise so I may switch it off, but I am still curious enough to leave it on for a bit longer.
